test/js/loan_store_test.js

```diff
@@ -1,35 +1,303 @@
 var chai = require('chai');
 var expect = chai.expect;
 var loanStore;
+var scenarioStore;
 var $;
 var sinon = require('sinon');
 var mortgageCalculations = require('../../src/static/js/modules/loan-comparison/mortgage-calculations.js');
 
+var api;
+
 
 describe('Loan store tests', function() {
     before(function () {
         $ = require('jquery');
         loanStore = require('../../src/static/js/modules/loan-comparison/stores/loan-store.js');
+        scenarioStore = require('../../src/static/js/modules/loan-comparison/stores/scenario-store.js');
+        api = require('../../src/static/js/modules/loan-comparison/api.js');
     });
     
+    describe('init', function() {
+        it('should reset all loans when init', function() {
+            var initStub = sinon.stub(loanStore, 'resetAllLoans');
+
+            loanStore.init();
+
+            sinon.assert.calledOnce(loanStore.resetAllLoans);
+            
+            loanStore.resetAllLoans.restore();
+        });
+    });
+
     describe('reset all loans', function() {
+        it('should reset all loans when there are no loans', function() {
+
+            var getScenarioStub = sinon.stub(scenarioStore, 'getScenario');
+            var updateDPConstStub = sinon.stub(loanStore, 'updateDownpaymentConstant');
+            var resetLoanStub = sinon.stub(loanStore, 'resetLoan');
```

Comment:
according to http://jaketrent.com/post/sinon-spies-vs-stubs/ you want to use a spy instead of a stub here